test/wasm-contracts/src/processor.ts

```diff
@@ -0,0 +1,40 @@
+import {SubstrateProcessor} from "@subsquid/substrate-processor"
+import {TypeormDatabase} from "@subsquid/typeorm-store"
+import {assertNotNull} from "@subsquid/util-internal"
+import {Flip} from "./model"
+import {Abi} from "@polkadot/api-contract"
+import flipperMetadata from "./metadata.json"
+import assert from "assert"
+
+
+const processor = new SubstrateProcessor(new TypeormDatabase('wasm_contracts'))
+
+
+processor.setDataSource({
+    chain: assertNotNull(process.env.CHAIN_NODE),
+    archive: assertNotNull(process.env.ARCHIVE)
+})
+
+
+processor.addContractsEventHandler(
+    "0xde8f2aa373e804aa57dd18b16de1249477375e205ec6c8a8bb129a5c4067a81c",
+    async ctx => {
+        let { event, block } = ctx.substrate;
+
+        let abi = new Abi(flipperMetadata)
+        let bytes = Buffer.from(ctx.data.slice(2), 'hex')
+        let decoded = abi.decodeEvent(bytes)
```

Comment:
Feeling very uneasy about the dependency of polkadot.js here as even they admit it as 'unstable': https://github.com/polkadot-js/api/blob/8649c17dd7055a82a719a3d9f489df6d406c4fd1/packages/api-contract/src/Abi/index.ts#L121

Reply:
it's just a test project and processor doesn't require this lib
a decision which abi lib to use is up to mapping developers